### ISSUE TITLE
Allow appToken deployment to be able to fetch the build logs

### ISSUE THREAD

**Issue by githubsaturn** [caprover/caprover#1182]
See here for the context: https://github.com/caprover/caprover/issues/1181

TLDR: currently the appToken based deployment aren't able to fetch the build logs. In some cases, it is useful to fetch the build logs for appToken deployments.

**Comment by alteoxCOM** [comment]
Hi, 
is there any workaround to make build logs available without giving developers master password of caprover?
Does caprover have a log file per app or one global log with all build log of all apps?
Maybe one could make this file accessible?

**Comment by githubsaturn** [comment]
No, the build logs aren't available via a file.

**Comment by tms0** [comment]
Hi @githubsaturn,

Do you plan to make this change in a future release ? I would like to use appToken deployment in my CI/CD but I really need to know if the deployment succeeds or not.

Thanks

**Comment by githubsaturn** [comment]
No, but we'll have the webhooks in the next version - it can be used for failed builds. 

Also, for CI/CD, you shouldn't need to fetch the build logs. The preferred way to is to build your image on your CI/CD pipeline and just send the built artifact to your CapRover instance.

**Comment by tms0** [comment]
How do you send the built artifact ? The only solution I see is to use a Docker registry, this is something I prefer to avoid. And I like the idea of building the Docker image on my machine, I would like to use a collaborative CI/CD platform where the resources are very limited.

**Comment by githubsaturn** [comment]
Yes, using a Docker registry. Building on a remote machine saves your main server from the high CPU usage of build time.

https://caprover.com/docs/ci-cd-integration.html

**Comment by tms0** [comment]
IMHO, for simple usecases like deploying static apps ([as describe here](https://caprover.com/docs/recipe-deploy-create-react-app.html)), using a registry adds a lot of overhead.

If you don't want to make this change for security concerns, maybe we can imagine something where the CLI wait until the deployment succeds or not and exits with the expected return code, with no logs ?
